devicetypes/heltun/heltun-tps05-switch.src/heltun-tps05-switch.groovy

```diff
@@ -0,0 +1,680 @@
+			)
+		}
+	}
+}
+
+def checkParam() {
+	boolean needConfig = false
+	parameterMap().each {
+		if (state."$it.name" == null || state."$it.name".state == "defNotConfigured") {
+			state."$it.name" = [value: it.default as Integer, state: "defNotConfigured"]
+			needConfig = true
+		}
+		if (settings."$it.name" != null && (state."$it.name".value != settings."$it.name" as Integer || state."$it.name".state == "notConfigured")) {
+			state."$it.name".value = settings."$it.name" as Integer
+			state."$it.name".state = "notConfigured"
+			needConfig = true
+		}
+	}
+	if ( needConfig ) {
+		configParam()
+	}
+}
+
+private configParam() {
+	def cmds = []
+	for (parameter in parameterMap()) {
+		if ( state."$parameter.name"?.value != null && state."$parameter.name"?.state in ["notConfigured", "defNotConfigured"] ) {
+			cmds << zwave.configurationV2.configurationSet(scaledConfigurationValue: state."$parameter.name".value, parameterNumber: parameter.paramNum, size: parameter.size).format()
+			cmds << zwave.configurationV2.configurationGet(parameterNumber: parameter.paramNum).format()
+			break
+		}
+	}
+	if (cmds) {
+		runIn(5, "checkParam")
+		sendHubCommand(cmds,500)
+	}
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.sensormultilevelv5.SensorMultilevelReport cmd) {
+	def map = [:]
+	def roomTemperature = 1
+	def humidity = 5
+	def illuminance = 3
+	def localScale = getTemperatureScale() //HubScale
+	def deviceScale = (cmd.scale == 1) ? "F" : "C" //DeviceScale
+	def child = childDevices?.find {channelNumber(it.deviceNetworkId) == 1 }
+	if (roomTemperature == cmd.sensorType) {
+		def deviceTemp = cmd.scaledSensorValue
+		def scaledTemp = (deviceScale == localScale) ? deviceTemp : (deviceScale == "F" ? roundC(fahrenheitToCelsius(deviceTemp)) : celsiusToFahrenheit(deviceTemp).toDouble().round(0).toInteger())
+		map.name = "temperature"
+		map.value = scaledTemp
+		map.unit = localScale
+		sendEvent(map)
+	} else if (humidity == cmd.sensorType) {
+		map.name = "humidity"
+		map.value = cmd.scaledSensorValue.toInteger()
+		map.unit = "%"
+		sendEvent(map)
+	} else if (illuminance == cmd.sensorType) {
+		map.name = "illuminance"
+		map.value = cmd.scaledSensorValue
+		sendEvent(map)
+	}
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.configurationv2.ConfigurationReport cmd) {
+	def parameter = parameterMap().find( {it.paramNum == cmd.parameterNumber } ).name
+	if (state."$parameter".value == cmd.scaledConfigurationValue){
+		state."$parameter".state = "configured"
+	}
+	else {
+		state."$parameter".state = "error"
+	}
+	configParam()
+}
+
+def updated() {
+	if (childDevices && device.label != state.oldLabel) {
+		childDevices.each {
+			def newLabel = getChildName(channelNumber(it.deviceNetworkId))
+			it.setLabel(newLabel)
+		}
+		state.oldLabel = device.label
+	}
+	initialize()
+}
+
+def initialize() {
+	runIn(3, "checkParam")
+}
+
+def installed() {
+	def numberOfButtons = 5
+	state.numberOfButtons = numberOfButtons
+	state.oldLabel = device.label
```

Comment:
Why do even need to use `state.oldLabel`?

Reply:
Was it ever needed? I don't recall that.

Reply:
@PKacprowiczS You are correct, it is not needed.
I removed it

Reply:
@Sarkis008 Did You commit those changes ? It's still there

Reply:
here it is. 
Did I do anything wrong ?
https://github.com/SmartThingsCommunity/SmartThingsPublic/blob/0783e698e9fa5c72b524d7e851adce75201c304c/devicetypes/heltun/heltun-tps05-switch.src/heltun-tps05-switch.groovy#L151